scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixArguments.java

```diff
@@ -205,4 +205,14 @@ ScalafixArguments withToolClasspath(
      */
     ScalafixError[] run();
 
+    /**
+     *
+     * Compared to {@link #run()}, it doesn't write the diff to the requested file,
+     * but instead return the list of patches and diagnostics, which can be written to file using
+     * {@link ScalafixResult#writeResult()}}.
+     * withCallback doesn't have any effect, all diagnostics are directly stored in ScalafixResult
+     *
+     * @return ScalafixResult
+     */
+    ScalafixResult runAndReturnResult();
```

Comment:
To echo the comment below about `ScalafixResult`, this could be something like
```suggestion
    ScalafixEvalution evaluate();
```

Reply:
good name